common.mk

```diff
@@ -361,6 +362,26 @@ edk2-clean-common:
 	$(MAKE) -j1 -C $(EDK2_PATH)/BaseTools clean && \
 	$(call edk2-call) cleanall
 
+################################################################################
+# U-Boot
+################################################################################
+UBOOT_COMMON_FLAGS ?= CROSS_COMPILE=$(CROSS_COMPILE_NS_KERNEL)
+
+$(UBOOT_PATH)/.config: $(UBOOT_DEFCONFIG_FILES)
+	cd $(UBOOT_PATH) && \
+                scripts/kconfig/merge_config.sh $(UBOOT_DEFCONFIG_FILES)
+
+.PHONY: uboot-defconfig
+uboot-defconfig: $(UBOOT_PATH)/.config
+
+.PHONY: uboot-common
+uboot-common: uboot-defconfig
+	$(MAKE) -C $(UBOOT_PATH) $(UBOOT_COMMON_FLAGS)
+
+.PHONY: uboot-clean-common
+uboot-clean-common:
+	$(MAKE) -C $(UBOOT_PATH) $(UBOOT_COMMON_FLAGS) distclean
+
```

Comment:
1. I think all platforms should use these targets as soon as they are introduced. How bout one refactoring commit with the changes to `common.mk` and all platform files ("Move u-boot* targets to common.mk"), then one separate commit for the new QEMUv8 stuff?
2. Why `uboot*` when `qemu.mk` already uses the spelling `u-boot*`?

Reply:
> 1. I think all platforms should use these targets as soon as they are introduced. How about one refactoring commit with the changes to `common.mk` and all platform files ("Move u-boot* targets to common.mk"), then one separate commit for the new QEMUv8 stuff?

Can I just add u-boot target in QEMUv8 file for now as I did in earlier PR. I need this added to get the xen workflow added. Then I can work on a separate PR to move things to common.mk and fix all platforms. Would that work ?

> 2. Why `uboot*` when `qemu.mk` already uses the spelling `u-boot*`?

 No particular reason. I actually didn't see the qemu.mk. I can change to u-boot for consistency.

Reply:
> Then I can work on a separate PR to move things to common.mk and fix all platforms. Would that work ?

Yes, totally acceptable.

> No particular reason. I actually didn't see the qemu.mk. I can change to u-boot for consistency.

Yes, please. No big deal, TBH I would have chosen `uboot` too, but consistency is more important.

Reply:
> > No particular reason. I actually didn't see the qemu.mk. I can change to u-boot for consistency.
> 
> Yes, please. No big deal, TBH I would have chosen `uboot` too, but consistency is more important

I could be the one to blame for that. I think I've tried to use the correct spelling. But if it was today, I probably would've went with `uboot` as well. (and I'm glad we use `optee_xyz` for gits etc and not `op-tee_xyx`).